cmd/gateclient/client.go

```diff
@@ -161,14 +139,14 @@ func (m *ApiMeta) Process(args []string) ([]string, error) {
 	// Api client initialization.
 	err = m.Authenticate()
 	if err != nil {
-		m.Ui.Error(fmt.Sprintf("OAuth2 Authentication failed."))
-		return args, err
+		util.UI.Error(fmt.Sprintf("OAuth2 Authentication failed."))
+		return nil, err
 	}
 
 	client, err := m.InitializeClient()
 	if err != nil {
-		m.Ui.Error(fmt.Sprintf("Could not initialize http client, failing."))
-		return args, err
+		util.UI.Error(fmt.Sprintf("Could not initialize http client, failing."))
```

Comment:
May be worth it at some point to create a `UI.Errorf()` function at some point to clean up the `fmt.Sprintf()s`. Not necessary in this PR though.